kafka-rest/src/test/java/io/confluent/kafkarest/v2/LoadTest.java

```diff
@@ -0,0 +1,172 @@
+        }
+
+        private List<ConsumerRecord<byte[], byte[]>> referenceRecords() {
+            return Arrays.<ConsumerRecord<byte[], byte[]>>asList(
+                    new BinaryConsumerRecord(topicName, "k1".getBytes(), "v1".getBytes(), 0, latestOffset - 3),
+                    new BinaryConsumerRecord(topicName, "k2".getBytes(), "v2".getBytes(), 0, latestOffset - 2),
+                    new BinaryConsumerRecord(topicName, "k3".getBytes(), "v3".getBytes(), 0, latestOffset - 1)
+            );
+        }
+
+        private void schedulePoll() {
+            consumer.schedulePollTask(new Runnable() {
+                @Override
+                public void run() {
+                    consumer.addRecord(new org.apache.kafka.clients.consumer.ConsumerRecord<>(topicName, 0, latestOffset, "k1".getBytes(), "v1".getBytes()));
+                    consumer.addRecord(new org.apache.kafka.clients.consumer.ConsumerRecord<>(topicName, 0, latestOffset + 1, "k2".getBytes(), "v2".getBytes()));
+                    consumer.addRecord(new org.apache.kafka.clients.consumer.ConsumerRecord<>(topicName, 0, latestOffset + 2, "k3".getBytes(), "v3".getBytes()));
+                    latestOffset += 3;
+                }
+            });
+        }
+    }
+
+    /**
+     * Test continuous reads for 60 seconds. 50 consumers, separated in 5 consumer groups.
+     */
+    @Test
+    public void testMultipleConsumerMultipleGroups() throws Exception {
+        Properties props = new Properties();
+        props.setProperty(KafkaRestConfig.BOOTSTRAP_SERVERS_CONFIG, "PLAINTEXT://hostname:9092");
+        props.setProperty(KafkaRestConfig.CONSUMER_MAX_THREADS_CONFIG, "-1");
+        props.setProperty(KafkaRestConfig.CONSUMER_REQUEST_TIMEOUT_MS_CONFIG, Long.toString(requestTimeoutMs));
+        config = new KafkaRestConfig(props, new SystemTime());
+        consumerManager = new KafkaConsumerManager(config, consumerFactory);
+        List<ConsumerTestRun> consumers = new ArrayList<>();
+        for (int group = 0; group < 5; group++) {
+            for (int i = 0; i < 10; i++) {
+                consumers.add(new ConsumerTestRun(
+                        new MockConsumer<>(OffsetResetStrategy.EARLIEST, Integer.toString(group))));
+            }
+        }
+        capturedConsumerConfig = Capture.newInstance();
+        Properties properties = EasyMock.capture(capturedConsumerConfig);
+        IExpectationSetters<Consumer> a = EasyMock.expect(consumerFactory.createConsumer(properties));
+        Method andReturnInstance = a.getClass().getMethod("andReturn", Object.class);
+        for (ConsumerTestRun run: consumers) {
+            andReturnInstance.invoke(a, run.consumer);
+        }
+        EasyMock.replay(consumerFactory);
+
+        for (ConsumerTestRun run: consumers) {
+            run.bootstrap();
+        }
+
+        for (ConsumerTestRun run: consumers) {
+            run.read();
+        }
+        for (int i = 0; i < 60; i++) {
+            awaitRead();
```

Comment:
I'm wondering if there's any way we could add some variability here. To simulate real load, the requests would be arriving at different intervals rather than all at once.